aries_cloudagent/connections/base_manager.py

```diff
@@ -739,6 +739,21 @@ async def get_connection_targets(
             targets = await self.fetch_connection_targets(connection)
         return targets
 
+    async def clear_connection_targets_cache(self, connection_id: str):
+        """Clear the connection targets cache for a given connection ID.
+
+        Historically, connections have not been updatable after the protocol
+        completes. However, with DID Rotation, we need to be able to update
+        the connection targets and clear the cache of targets.
+        """
+        # TODO it would be better to include the DIDs of the connection in the
+        # target cache key This solution only works when using whole cluster
+        # caching or have only a single instance with local caching
```

Comment:
I think we should discuss this point; if we don't make this change now, there can be unexpected behavior in clusters. Additionally, once the rotation is completed, there is no way to handle any potentially out of order messages. If we did make this change, the cached value for the previous DIDs would be around until the cache expires them. There would still be issues in clustered environments without whole cluster caching, though.

Reply:
TBH I'm not entirely familiar with how the different caching mechanisms work.

Reply:
Yeah, I'm feeling like the right option right now is probably to log an issue to address handling potentially out of order messages received after a rotation. It's an unlikely edge case and I think it will be more valuable to keep this PR moving for now.